server/apps/immich/src/api-v1/config/config.controller.ts

```diff
@@ -0,0 +1,34 @@
+import { BadRequestException, Body, Controller, Get, Put, UseGuards, ValidationPipe } from '@nestjs/common';
+import { ConfigService } from './config.service';
+import { ApiBearerAuth, ApiTags } from '@nestjs/swagger';
+import { JwtAuthGuard } from '../../modules/immich-jwt/guards/jwt-auth.guard';
+import { AdminRolesGuard } from '../../middlewares/admin-role-guard.middleware';
+import { AdminConfigResponseDto } from './response-dto/admin-config-response.dto';
+import { SetAdminConfigDto } from './dto/set-admin-config';
+import { AdminConfigKey } from '@app/database/entities/admin-config.entity';
+
+@UseGuards(JwtAuthGuard)
+@UseGuards(AdminRolesGuard)
+@ApiTags('Admin Config')
+@ApiBearerAuth()
+@Controller('config')
+export class ConfigController {
+  constructor(private readonly adminConfigService: ConfigService) {}
+
+  @Get('/admin')
+  getAdminConfig(): Promise<AdminConfigResponseDto> {
+    return this.adminConfigService.getAllConfig();
+  }
+
+  @Put('/admin')
+  async putAdminConfig(@Body(ValidationPipe) body: SetAdminConfigDto): Promise<AdminConfigResponseDto> {
+    if (
+      body.config.filter((entry) => {
```

Comment:
This is all validation that can be accomplished via the dto + validation pipe.

Reply:
I tried using the validation pipe but couldn't get it to work properly, happy for you to update this to use the validation pipe so I can see how it's supposed to be used. My experience with using Nest is limited to this project so I am still learning how it all works as it's pretty different to everything else i've used before 😄